### ISSUE TITLE
Pyright complains type issues when use chain method

### ISSUE THREAD

**Issue by Gnosnay** [dgilland/pydash#228]
pyright version: 1.1.370
pydash: 8.0.1

how to reproduce:
`demo.py` is as follows:
```python
from pydash import chain

chain([1, 2, 3, 4]).map(lambda x: x * 2).sum().value()
```
execute
```shell
pyright demo.py
```

message:
```
Type of "map" is partially unknown
    Type of "map" is "Overload[(iteratee: (int) -> T2@map_) -> Chain[List[T2@map_]], (iteratee: (int, int) -> T2@map_) -> Chain[List[T2@map_]], (iteratee: (int, int, List[int]) -> T2@map_) -> Chain[List[T2@map_]], (iteratee: int | str | List[Unknown] | Tuple[Unknown, ...] | Dict[Unknown, Unknown] | None = None) -> Chain[List[Unknown]]]" (reportUnknownMemberType)
```

May i know how to fix this problem?

**Comment by DeviousStoat** [comment]
Hey @Gnosnay thanks for the report. I linked a PR that should solve this :)
Note that it seems that you are using pyright in strict mode which enables a lot of strict diagnostic rules and this library uses mypy so you might see other type issues like this one. However I am really willing to make it support strict pyright in an incremental fashion so don't hesitate to create other issues if you stumble upon other pyright complains.

**Comment by Gnosnay** [comment]
> Hey @Gnosnay thanks for the report. I linked a PR that should solve this :) Note that it seems that you are using pyright in strict mode which enables a lot of strict diagnostic rules and this library uses mypy so you might see other type issues like this one. However I am really willing to make it support strict pyright in an incremental fashion so don't hesitate to create other issues if you stumble upon other pyright complains.

hi @DeviousStoat the PR looks good. thx for your help

**Comment by dgilland** [comment]
Fix for this has been release with [v8.0.2](https://pypi.org/project/pydash/8.0.2/).
